webpack.config.js

```diff
@@ -2,21 +2,44 @@
 const WebpackNotifierPlugin = require("webpack-notifier");
 const webpack = require("webpack");
 const path = require("path");
-const yaml = require("yamljs");
 const absolute = (relPath) => path.join(__dirname, relPath);
+const webpackConfig = require("./bin/webpack-config");
 
 const srcPath = absolute("./content-src/main.js");
 const outputDir = absolute("./data/content");
 const outputFilename = "bundle.js";
 
-let config = yaml.load("config.default.yml");
-try {
-  // Load user config if it exists
-  config = Object.assign({}, config, yaml.load("config.yml"));
-} catch (e) {} // eslint-disable-line no-empty
+const IS_PRODUCTION = process.env.NODE_ENV === "production";
+const IS_TEST = process.env.NODE_ENV === "test";
+
+let plugins =  [
+  new WebpackNotifierPlugin(),
+  new webpack.DefinePlugin(webpackConfig(IS_PRODUCTION))
+];
+
+if (!IS_TEST) {
+  plugins.push(new webpack.optimize.CommonsChunkPlugin("vendor", "vendor.bundle.js"));
+}
+
+if (IS_PRODUCTION) {
+  plugins = plugins.concat([
+    new webpack.optimize.OccurenceOrderPlugin(),
+    new webpack.optimize.UglifyJsPlugin({
+      test: /vendor/
+    }),
```

Comment:
We're getting a lot of noise from the vendor bundle. Not sure if we want to silence the warnings using something like:
```
    new webpack.optimize.UglifyJsPlugin({
      test: /vendor/,
      compress: {
        warnings: false
      }
    }),
```